posts/2022-09-15-const-eval-safety-rule-revision.md

```diff
@@ -0,0 +1,313 @@
+---
+layout: post
+title: Const Eval Safety Rules
+author: Felix Klock
+description: "Various ways const-eval can change between Rust versions"
+team: The Compiler Team <https://www.rust-lang.org/governance/teams/compiler>
+---
+
+In a recent Rust issue ([#99923][]), a developer noted that the upcoming
+1.64-beta version of Rust had started signalling errors on their crate,
+[`icu4x`][icu4x]. The `icu4x` crate makes heavy use of *const-eval*: Rust code
```

Comment:
I might be over-indexing on this, but I am concerned people will read the first few sentences and assume there is some problem with Rust's const-eval in general. I think it would be better to mention that `icu4x` uses `const+unsafe` code here rather than saying "they use a lot of const-eval" as that isn't really the problem.

Reply:
Hopefully addressed by 649bc11a192ec3a817f7406fb94f588eac281479